src/linter.rs

```diff
@@ -43,30 +43,35 @@ impl Context {
   pub(crate) fn add_diagnostic(
     &mut self,
     span: Span,
-    code: impl Into<String>,
-    message: impl Into<String>,
+    code: impl ToString,
+    message: impl ToString,
   ) {
-    let diagnostic = self.create_diagnostic(span, code, message, None);
+    let diagnostic =
+      self.create_diagnostic(span, code.to_string(), message.to_string(), None);
     self.diagnostics.push(diagnostic);
   }
 
   pub(crate) fn add_diagnostic_with_hint(
     &mut self,
     span: Span,
-    code: impl Into<String>,
-    message: impl Into<String>,
-    hint: impl Into<String>,
+    code: impl ToString,
+    message: impl ToString,
+    hint: impl ToString,
   ) {
-    let diagnostic =
-      self.create_diagnostic(span, code, message, Some(hint.into()));
+    let diagnostic = self.create_diagnostic(
+      span,
+      code.to_string(),
+      message.to_string(),
+      Some(hint.to_string()),
+    );
     self.diagnostics.push(diagnostic);
   }
 
   fn create_diagnostic(
     &self,
     span: Span,
-    code: impl Into<String>,
-    message: impl Into<String>,
+    code: String,
+    message: String,
```

Comment:
This refactor is inspired by https://www.possiblerust.com/pattern/non-generic-inner-functions, in which the author says it is better to decrease generic functons as few as possible for faster compile time and less binary size.

Reply:
I'm pretty sure this won't matter much as most of the time is spent compiling swc and its macros anyway :)

Reply:
Sounds right, to be honest I just wanted to practice it, so I don't have a strong will to do it. I will revert :)